### PR TITLE
Improved diff syntax highlighting fix

### PR DESCRIPTION
**Pull request by mrsdizzie**
Make previous fix from #12238 more robust since I saw a case where a diff changes only a single character in a chroma class instead of the entire thing. Add another more complicated test to match.

Thanks @jolheiser for noticing

Before:
Screen Shot 2020-08-07 at 2 03 01 PM
After:
Screen Shot 2020-08-07 at 2 03 11 PM

### CODE REVIEW

File: services/gitdiff/gitdiff.go [hunk 1]
```diff
@@ -180,55 +181,61 @@ var (
 	removedCodePrefix = []byte(`<span class="removed-code">`)
 	codeTagSuffix     = []byte(`</span>`)
 )
+var re = regexp.MustCompile(`<span class="[a-z]*$`)
```
Comment (lafriks) [inline services/gitdiff/gitdiff.go]: It should be named more specific as it is global variable

Comment (mrsdizzie) [reply services/gitdiff/gitdiff.go]: right : ) fixed

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Review by bagasme — COMMENTED** [review]
ok

**Review by lafriks — APPROVED** [review]

**Pull request merged**